Asterix/Optical_System_functions.py

```diff
@@ -1899,29 +2028,72 @@ def voltage_to_phase(self, actu_vect, einstein_sum=False):
     def create_DM_basis(self, basis_type='actuator'):
         """ --------------------------------------------------
         Create a DM basis.
-        TODO do a sine / cosine basis and a
-        TODO do a zernike basis
+        TODO do a zernike basis ?
+
+        AUTHOR: Johan Mazoyer
 
-        Parameters:
+        Parameters
         ----------
-        DM: a DM object (Optical System)
-        basis_type: string, default 'actuator' the type of basis
+        basis_type: string, default 'actuator' 
+            the type of basis. 'fourier' or 'actuator'
 
-        Return:
+        Returns
         ------
-        a 2d numpy array [Size basis, Number act in the DM]
+        basis: 2d numpy array 
+            basis [Size basis, Number of active act in the DM]
+
+
         -------------------------------------------------- """
-        active_and_in_pup = [
-            value for value in self.active_actuators
-            if value in self.WhichInPupil
-        ]
-        active_and_in_pup.sort()
 
         if basis_type == 'actuator':
+            active_and_in_pup = [
+                value for value in self.active_actuators
+                if value in self.WhichInPupil
+            ]
+            active_and_in_pup.sort()
+
             basis_size = len(active_and_in_pup)
-            basis = np.zeros((basis_size, self.number_act), dtype=int)
+            basis = np.zeros((basis_size, self.number_act))
             for i in range(basis_size):
                 basis[i][active_and_in_pup[i]] = 1
+
+        elif basis_type == 'fourier':
+            start_time = time.time()
+            activeact = [value for value in self.active_actuators]
+
+            sqrtnbract = int(np.sqrt(self.total_act))
```

Comment:
assume DM is square. Le préciser dans le HOWTO et dans le fichier d'init.

Reply:
en fait les DMs sont tous Square, pour le BMC 34x34 je commence par dire qu'il fait 34x34 et puis j'ai un fichier avec les actu non actifs. Donc cette valeur est aussi défini pour les non squares